airflow/models.py

```diff
@@ -2473,6 +2476,7 @@ def __init__(
             run_as_user=None,
             task_concurrency=None,
             executor_config=None,
+            do_xcom_push=True,
```

Comment:
True by default?

Reply:
Since this is a global flag, it should be true by default to not change the behaviour of all existing operators that push.

I think?

Reply:
XCOM feature if not used should be disabled by default as in base operators, users should declare explicitly the use of XCOM so that database is not filled with useless data.

For backwards compatibility, contrib operators, which currently push to XCOM by default,  should be updated to explicitly activate XCOM. So that other contrib operators will keep the current behavior of not using XCOM.

Reply:
if i was rewriting airflow from scratch, i would agree with you @itscaro. 
but in this instance, were airflow is looking to mature a bit, i agree with @ashb; we should preserve bw compat where possible.

Reply:
I agree with @ashb. For backward compatibility, it should be true by default.

Reply:
Airflow currently operates as If this value has True by default -- if I write a custom operator and `return "foo"` from it that will make it to XCom. This is the behaviour we need to keep.

Reply:
I missed the case about operators outside of this repository.  We might start to make it defaults to False for operators in this repository, for other modules the default is True for backwards compatibility until a major version?

To have to disable XCOM in each operator call might be something not wanted.

Reply:
@itscaro You could set it in the default_args then.